### ISSUE TITLE
Add game shader(?) cache / Fix new effects lag?

### ISSUE THREAD

**Issue by FatheredPuma81** [jpd002/Play-#1115]
No clue if this is feasible, worth implementing, or if there is a better way of solving this issue, but some games such as Kingdom Hearts 2 would benefit greatly from some sort of shader/effects cache that persists through emulator restarts. Would help alleviate the very poor performance on game start.
See 0:40 in video if you don't understand what I'm talking about. Game lags heavily when new effects are used.

W10
Ryzen 5 3600
GTX 1080

https://www.youtube.com/watch?v=FCyWqgxBL-I

**Comment by jpd002** [comment]
Yep, this is something very annoying that needs fixing. I think it's the JIT that's taking a noticeable amount of time to compile code blocks that are encountered when new enemies/areas/effects are being shown. 

There's lots of space for improvement in the JIT, I did a first pass last month or so that cut the compile time by at least an half and I'm pretty sure it could be reduced even further.

Shader compilation could also be causing some lag, but I really think the JIT is the main culprit here.

**Comment by FatheredPuma81** [comment]
I'm completely out of my depth, so bear with me I'm just throwing out "ideas". If it's only tied to visuals couldn't it be asynchronous? Kind of like what RPCS3 did with their shaders? https://www.youtube.com/watch?v=s7nFjhEf55U

Also. Play! seems to be improving quite a lot.
